wordpress/wp-content/plugins/cds-wpml-mods/resources/js/components/PageSelect.js

```diff
@@ -0,0 +1,159 @@
+const { SelectControl } = wp.components;
+const { useState, useEffect } = wp.element;
+const { select, subscribe } = wp.data;
+const { __ } = wp.i18n;
+
+import { getData, sendData } from '../util/fetch.js';
+
+export const PageSelect = () => {
+    const { isSavingPost } = select( 'core/editor' );
+
+    const emptyPage = { is_translated: null, value: 0, label: __("None", "cds-wpml-mods") };
+    // get the current post id
+    const postID = select("core/editor").getCurrentPostId();
+    const type = select("core/editor").getCurrentPostType();
+
+    const hintTexts = {
+        'empty': () => `${__('No translation assigned for this', "cds-wpml-mods")} ${type}.`,
+        'untranslated': (label) => `“${label}” ${__('will be assigned as translation for this', "cds-wpml-mods")} ${type}.`,
+        'translated': (label) => `${__('⚠️ This will unlink the existing translation for', "cds-wpml-mods")} “${label}”.`,
+        'translated_post_id': (label) => `“${label}” ${__('is the current translation for this', "cds-wpml-mods")} ${type}.`,
+    }
+
+    const [isLoading, setIsLoading] = useState(true);
+    const [post, setPost] = useState();
+    const [pages, setPages] = useState([emptyPage]);
+    const [page, setPage] = useState(emptyPage);
+    const [hintText, setHintText] = useState(hintTexts['empty']());
+    const [isSavingProcess, setSavingProcess] = useState(false);
+
+    /* Triggers when the "publish" or "update" button is clicked */
+    /* https://github.com/WordPress/gutenberg/issues/17632#issuecomment-1153888435 */
+    subscribe(() => {
+        if (isSavingPost()) {
+            setSavingProcess(true);
+        } else {
+            setSavingProcess(false);
+        }
+    });
+
+    const updatePost = async () => {
+        setIsLoading(true)
+        const response = await sendData(`cds/wpml/posts/${postID}/translation`, {translationId: page.value})
+
+        if (response.ID) {
+            setPost(response)
+            setIsLoading(false)
+        }
+    };
+
+    const sortPages = (pages) => {
+        const sortByPostTitle = (p1, p2) => (p1.post_title > p2.post_title) ? 1 : -1;
+
+        const untranslatedPages = pages.filter(p => !p.is_translated).sort(sortByPostTitle)
+        const translatedPages = pages.filter(p => p.is_translated).sort(sortByPostTitle)
+
+        // return all pages in one array: untranslated pages first, sorted by title, then translated pages sorted by title
+        return [...untranslatedPages, ...translatedPages]
+    }
+
+    useEffect(() => {
+        const getPost = async (postID) => {
+            const response = await getData(`cds/wpml/posts/${postID}/translation`);
+
+            if (response.ID) {
+                setPost(response)
+            }
+        }
+
+        getPost(postID);
+    }, [postID]);
+
+    useEffect(() => {
+        const getPages = async (post) => {
```

Comment:
Thinking it would be better to pass the post attributes we need for this function 

I think that will help prevent refetch / loading again when you  move between the page / block tabs


<img width="334" alt="Screen Shot 2022-07-29 at 9 58 42 AM" src="https://user-images.githubusercontent.com/62242/181777124-dda68b01-301e-4a1f-befa-ef7579e0217e.png">

Right now if you swap between them there's a fetch happening but there doesn't need to be.

Reply:
Ah, I wasn't sure why that was happening, but you're right, there's no reason it needs to.